src/cli/java/org/commcare/util/screen/EntityDetailSubscreen.java

```diff
@@ -46,21 +49,28 @@ public EntityDetailSubscreen(int currentIndex, Detail detail, EvaluationContext
                 dataTemporary.add(data);
                 headersTemporary.add(createHeader(field, subContext));
                 rowTemporary.add(createRow(field, subContext, data));
+                stylesTemporary.add(createStyle(field));
```

Comment:
@dannyroberts this is the important change - moving the Style[] accumulation inside this `if` statement that only adds when the value of the detail is non-empty. Previously we were using the `EntityScreen`'s built in `getStyles` function which didn't perform this check.

Reply:
Ah nice, makes sense!